### PR TITLE
Python 3.8 and 3.9 compatibility

### PR DESCRIPTION
**Pull request by HugoFara**
This branch brings proper compatibility to Python 3.8 and 3.9, but it also includes requirement updates, and code style improvements. 

## Description

The big chucks of the update are:
* Explicitly saying that Python 3.8/9 are supported. **Python 3.5 has been removed from tox.ini (EOL), Python 3.6 is untested.**
* Update of all requirements to their latest version available. Package should run fine without a lot of those updates, but their are very old version. In Actions, Ubuntu-16.04 was updated to Ubuntu-latest.
* Code was slightly adapted to the new versions. With this **this package is flake8 aware**. They have also been other code style conventions enforcement.

For convenience, Spyder projects characteristics were added to .gitignore as well as the .idea foldre related to PyCharm.

Bump version needs to be updated. We recommend a MINOR version update, because code behavior was left untouched, and no strong incompatibility was seen with previous requirement version. 

## Related Issue

This pull request solves #455 (Python 3.8/9 support). It is related to flake8 violations (issue #54 ).  

## Motivation and Context

Python 3.5 has reached EOL, Python 3.6 reaches EOL in [the end of 2021](https://devguide.python.org/#status-of-python-branches).

## How Has This Been Tested?

Codes has been tested with tox exclusively. 
Tests results are:
* Python 3.6/7: untested (compatibility shouldn't be broken).
* Python 3.8/9: 456 passed, 12 skipped, 5 warnings. 4 warnings are related to deprecated dependencies **in tox**. One is a UserWarning (not important). 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -20,13 +20,17 @@ stages:
   jobs:
   - job: 'Test'
     pool:
-      vmImage: 'Ubuntu-16.04'
+      vmImage: 'Ubuntu-latest'
     strategy:
         matrix:
           Python36:
```
Comment (ljvmiranda921) [inline azure-pipelines.yml:26]: Let's drop support for Python36 already :+1: Can remove this line

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py35, py36, py37, flake8
+envlist = py36, py37, py38, py39, flake8
```
Comment (ljvmiranda921) [inline tox.ini:2]: We can also remove Python 3.6 here

File: setup.py [hunk 3]
```diff
@@ -44,6 +44,8 @@
         "Topic :: Scientific/Engineering :: Mathematics",
         "Programming Language :: Python :: 3.6",
```
Comment (ljvmiranda921) [inline setup.py:45]: And lastly, here :+1:

**Review by ljvmiranda921 — COMMENTED** [review]
Hi @HugoFara , thanks a lot for this PR! I also appreciate that you updated the docstrings and made everything consistent. Let's drop Python 3.6 altogether, I highlighted areas in the codebase where we can remove that.

**Comment by HugoFara** [comment]
Hi! After many hours of debugging, I was just too curious to know if I achieved maintaining the py36 compatibility to remove it :smile: 

I thought you would have liked to wait for the official EOL, but no problem to do that now!

Eventually, the Python 3.6 CI looks broken due to server cache, but I'm quite confident it would have worked otherwise.

File: azure-pipelines.yml (hunk 1 above)
Comment (HugoFara) [reply azure-pipelines.yml]: Added in following push :ok_hand:

File: tox.ini (hunk 2 above)
Comment (HugoFara) [reply tox.ini]: Added in following push :ok_hand:

File: setup.py (hunk 3 above)
Comment (HugoFara) [reply setup.py]: Added in following push :ok_hand:

**Review by ljvmiranda921 — APPROVED** [review]

**Comment by ljvmiranda921** [comment]
Hi @HugoFara this is already approved! Let me merge within this week, just got a bit busy!

**Pull request closed without merging**